### PR TITLE
Make FFT analysis multi-threaded

### PR DESCRIPTION
**Pull request by reddvl1980**

### CODE REVIEW

File: test/integration/IntegrationTest.cpp [hunk 1]
```diff
@@ -223,6 +224,16 @@ api::ConferenceEndpointExtendedInfo IntegrationTest::getEndpointExtendedInfo(con
     return api::Parser::parseEndpointExtendedInfo(endpointRequest.getJsonBody());
 }
 
+size_t getNumWorkingThreads()
```
Comment (olofkallander) [inline test/integration/IntegrationTest.cpp:227]: This name in the same file creates confusion as there is already a getNumWorkerThreads.

File: test/integration/IntegrationTest.cpp [hunk 2]
```diff
@@ -256,26 +266,51 @@ void IntegrationTest::analyzeRecording(const std::vector<int16_t>& recording,
...
+    std::mutex fft_mutex;
 
-        std::vector<double> frequencies;
-        SampleDataUtils::listFrequencies(testVector, codec::Opus::sampleRate, frequencies);
-        for (size_t i = 0; i < frequencies.size() && i < 50; ++i)
-        {
-            if (std::find(frequencyPeaks.begin(), frequencyPeaks.end(), frequencies[i]) == frequencyPeaks.end())
+    for (size_t threadId = 0; threadId < numThreads; threadId++)
+    {
+        workers.push_back(std::thread([&frequencyPeaks, &recording, &fft_mutex, &logId, size, numThreads, threadId]() {
```
Comment (olofkallander) [inline test/integration/IntegrationTest.cpp:277]: I have read multiple times. I still do not like that we create a fat lambda starting at 3 indentation levels.
Break this out into a method and pass that one to the thread.

File: test/integration/IntegrationTest.cpp (hunk 1 above)
Comment (reddvl1980) [reply test/integration/IntegrationTest.cpp:227]: Agree.

File: test/integration/IntegrationTest.cpp [hunk 3]
```diff
@@ -256,26 +266,51 @@ void IntegrationTest::analyzeRecording(const std::vector<int16_t>& recording,
...
+                    testVector[x] =
+                        std::complex<double>(static_cast<double>(recording[x + cursor]), 0.0) / (256.0 * 128);
+                }
+
+                SampleDataUtils::fft(testVector);
+
+                std::vector<double> frequencies;
+                SampleDataUtils::listFrequencies(testVector, codec::Opus::sampleRate, frequencies);
+                {
+                    const std::lock_guard<std::mutex> lock(fft_mutex);
```
[the remark this answers is not in the document]
Comment (reddvl1980) [reply test/integration/IntegrationTest.cpp]: Fixed (with single threaded reduction in the end, do not think making reduction multithreaded is necessary).

**Pull request merged**